### DISCUSSION TITLE
Correct way of using specific port for HPC?

### DISCUSSION THREAD

**Discussion by Aariq** [mschubert/clustermq#284] [Q&A] [answered]
My institutions HPC only allows SSH connection without a password (using SSH key) on port 2222.  Right now, I have options set like so:

```
options(
  clustermq.scheduler = "ssh",
  clustermq.template = "ssh_clustermq.tmpl",
  clustermq.ssh.host = "-p 2222 user@host"
)
```
Is this the correct way to do this, or is there another option I should set to specify the port?  Currently this works, but it's inconsistent—usually on the first try to connect I get a timeout error instantly, but that's maybe a separate issue?

**Comment by mschubert** [comment, marked as answer]
You can set up how you connect via ssh using your `~/.ssh/config` (this can be a cmq-specific host):

```ssh
Host my-ssh-host-name
    Hostname host.ip.address.or.name
    User your_user
    Port 2222
    IdentityFile ...
```

And then use your defined host name to connect:

```r
options(
  clustermq.scheduler = "ssh",
  clustermq.ssh.host = "my-ssh-host-name"
)
```

**Reply by Aariq** [reply]
Thanks
